portal/extensions.py

```diff
@@ -154,3 +154,9 @@ def init_app(self, app):
 # Bable is used for i18n
 from flask_babel import Babel
 babel = Babel()
+
+@babel.localeselector
```

Comment:
Hate to sound so particular, but the extensions module should remain as small as possible.  It exists to work around compile time dependency lookups - where you need for example a database, or library instance variable for inclusion.

I would assume get_locale() might better be associated with the user object?